### PR TITLE
Update cilium-agent ServiceMonitor

### PR DESCRIPTION
**Pull request by nebula-it**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

Fixes: #issue-number

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ metadata:
 spec:
   selector:
     matchLabels:
-      k8s-app: cilium
+      app.kubernetes.io/name: cilium-agent
```
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:25]: Why this change? All other ServiceMonitor definitions also use "k8s-app", let's be consistent

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks for the PR! The Envoy DS change looks good to me, but the label change looks unintentional

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml (hunk 1 above)
Comment (nebula-it) [reply install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:25]: Sorry, I messed up the commit message while cleaning up my email. Here is the original commit message for this `Update the label to be more precise as exiting label selects both cilium-agent and hubble-peer services`

**Review by gandro — APPROVED** [review]

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml (hunk 1 above)
Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:25]: Could you update the commit message so it's correct again? Thanks

Comment (nebula-it) [reply install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:25]: Done

**Comment by nebula-it** [comment]
@gandro All done, can you please merge

**Comment by gandro** [comment]
Unfortunately it seems you pulled in a merge commit. The Cilium tree uses rebase - please remove the merge commit and rebase on main first

**Review by joamaki — APPROVED** [review]

**Comment by nebula-it** [comment]
@gandro Can you please help me merge this

**Comment by gandro** [comment]
I've restarted the failing (but required) CI jobs. Once CI is green, we can merge
